backend/controllers/addAppointmentDetails.js

```diff
@@ -41,11 +102,26 @@ const addAppointmentDetailsFactory = (bulkAppointmentService, oauthApi, logError
       const { activeCaseLoadId } = req.session.userDetails
       const { appointmentTypes, locations } = await getAppointmentTypesAndLocations(res.locals, activeCaseLoadId)
 
+      const { appointmentType, location, date, startTime, endTime, sameTimeAppointments, comments } =
+        req.session.data || {}
+
+      const startTimeHoursMinutes = startTime && extractHoursMinutes(startTime)
+      const endTimeHoursMinutes = endTime && extractHoursMinutes(endTime)
+
       res.render('addAppointmentDetails.njk', {
         title: 'Add appointment details',
         appointmentTypes,
         locations,
         homeUrl: notmEndpointUrl,
+        appointmentType,
+        location,
+        comments,
+        date,
+        startTimeHours: startTime && String(startTimeHoursMinutes.hours),
```

Comment:
you're already testing to see if `startTime` set above, so this is a duplicate.  If you use `format` instead then casting to `String` here won't be required.  Maybe `extractHoursMinutes` could cope with time not being set and then you could just use spread operator here for hours and minutes

Reply:
You're right, let me clean the mapping code up a bit.